### PR TITLE
Build: Add NodeJS script to retrieve the URL for an artifact in TeamCity

### PR DESCRIPTION
**Pull request by yuliyan**
#### Changes proposed in this Pull Request

* Add NodeJS script to retrieve the URL for an artifact in TeamCity.

#### Testing instructions

* Test with `node bin/get-teamcity-artifact-url --artifact="translate/calypso-strings.pot" --branch="update/migrate-translate-script-to-tc"`.

Related to 378-gh-Automattic/i18n-issues

### CODE REVIEW

**Comment by noahtallen** [comment]
_Accidentally posted this to the other PR first, sorry!_

I think we can make this even easier! TeamCity allows guest auth which lets us do some interesting things. For example, this URL will download the most recent successful calypso strings artifact associated with https://github.com/Automattic/wp-calypso/pull/60916.

https://teamcity.a8c.com/repository/download/calypso_Translate/.lastSuccessful/translate/calypso-strings.pot?guest=1&branch=update/migrate-translate-script-to-tc

I'm not sure what your use case is, but if the only purpose of the script is to download the most recent successful trunk artifact, you could just hardcode the above URL and just change branch=update/... to branch=trunk. If you need to parameterize more things, I would just make the script format the above URL. For example:

```js
const buildType = 'calypso_Translate';
const artifactName = 'translate/calypso-strings.pot';
const branchName = 'trunk';

const tcURL = `https://teamcity.a8c.com/repository/download/${ 
  buildType
}/.lastSuccessful/${
  artifactName
}?guest=1&branch=${
  branchName
}`;
```

And instead of using .lastSuccessful along with branch=, you could pass a build ID in its place: ...download/calypso_Translate/$buildID/$artifactName?guest=1

This type of URL works on my sandbox with curl, and on my local computer. But I'm not sure where you'll need to run it :)

It also took me a fair amount of exploration to figure out this format! Otherwise, it would have been tricky to get everyone to generate an auth token and add it to their sandbox. The ?guest=1 mode isn't quite as well documented either.

**Comment by yuliyan** [comment]
@noahtallen, Wow, that makes it a lot easier to use and makes the implementation a lot simpler.

[TeamCity's Docs advised using the REST API](https://www.jetbrains.com/help/teamcity/2021.2/patterns-for-accessing-build-artifacts.html), which is why I went straight to using it for the initial implementation. However, in this case there is indeed no need for that and also seems to not provide any benefits.

I've updated the PR to drop the unnecessary dependencies and use the URL pattern instead.

Thank you for the suggestion!

File: bin/get-teamcity-artifact-url [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+	}, {} );
+
+const args = {
+	branch: 'trunk',
+	buildType: 'calypso_Translate',
+	artifact: '',
+	...parsedArgs,
+};
+
+( async function () {
```
Comment (noahtallen) [inline bin/get-teamcity-artifact-url]: I don't think we need the async function here

File: bin/get-teamcity-artifact-url [hunk 2]
```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env node
+
+// This script will return a url of an artifact generated in the latest project build for the specified branch.
+// eg: node bin/get-teamcity-artifact-url --artifact="translate/calypso-strings.pot"
+
+const parsedArgs = process.argv
```
Comment (noahtallen) [inline bin/get-teamcity-artifact-url]: If you want, you could use `yargs` here -- I think we have it available in Calypso. It might not make things easier in this simple case, but I know it supports things like default values and whatnot

File: bin/get-teamcity-artifact-url [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+const args = {
+	branch: 'trunk',
+	buildType: 'calypso_Translate',
+	artifact: '',
+	...parsedArgs,
+};
+
+( async function () {
+	if ( args.artifact ) {
```
Comment (noahtallen) [inline bin/get-teamcity-artifact-url]: Definitely nitpicky, but I would rephrase this as:

```js
if ( ! args.artifact ) {
  process.exit( 1 )
}

const url = ...
```

**Review by noahtallen — COMMENTED** [review]
Nice, it's so much better!

File: bin/get-teamcity-artifact-url (hunk 2 above)
Comment (yuliyan) [reply bin/get-teamcity-artifact-url]: Thanks for pointing out yargs. I looked before if we had arguments parsing libraries already available in Calypso and somehow missed yargs 🤦. Again, pretty nice improvement.

File: bin/get-teamcity-artifact-url (hunk 3 above)
Comment (yuliyan) [reply bin/get-teamcity-artifact-url]: With the switch to `yargs` and the use of `.required()`, it's no longer needed to do the check.

**Review by tyxla — APPROVED** [review]
Tests well and the code looks great 🚢
